src/Acuminator/Acuminator.Analyzers/StaticAnalysis/EventHandlerModifier/EventHandlerModifierAnalyzer.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System.Collections.Immutable;
+using Acuminator.Analyzers.StaticAnalysis.PXGraph;
+using Acuminator.Utilities.Roslyn.Semantic.PXGraph;
+using Acuminator.Utilities.Roslyn.Semantic;
+using Microsoft.CodeAnalysis;
+using System.Linq;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Generic;
+
+namespace Acuminator.Analyzers.StaticAnalysis.PrivateEventHandlers
+{
+	public class EventHandlerModifierAnalyzer : PXGraphAggregatedAnalyzerBase
+	{
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+			ImmutableArray.Create(
+				Descriptors.PX1077_EventHandlersShouldBeProtectedVirtual,
+				Descriptors.PX1077_EventHandlersShouldNotBeExplicitInterfaceImplementations);
+
+		public override void Analyze(SymbolAnalysisContext context, PXContext pxContext, PXGraphEventSemanticModel graphExtension)
+		{
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			foreach (var handler in graphExtension.GetAllEvents().Where(graphEvent => graphEvent.Symbol.IsDeclaredInType(graphExtension.Symbol)))
+			{
+				context.CancellationToken.ThrowIfCancellationRequested();
+
+				var location = handler.Symbol.Locations.FirstOrDefault();
+
+				if (location == null)
+				{
+					continue;
+				}
+
+				if (handler.Symbol.IsOverride)
+				{
+					continue;
+				}
+
+				var properties = new Dictionary<string, string?>
```

Comment:
Please introduce string constants for diagnostic properties.
Also I would advice you to use common `DiagnosticProperties` if possible. For example, you can use `DiagnosticProperty.RegisterCodeFix` property and pass it to the code fix like this:
```cs
bool registerCodeFix = !handler.Symbol.IsOverride && 
                                    handler.Symbol.MethodKind != MethodKind.ExplicitInterfaceImplementation && 
                                    !handler.ImplementsInterface();
var properties = new Dictionary<string, string?>
{
   { DiagnosticProperty.RegisterCodeFix, registerCodeFix .ToString()  }
}
.ToImmutableDictionary();
```

Reply:
Done. I added one new string constant for the IsContainedTypeSealed. And also I reused RegisterCodeFix.